### PR TITLE
fix: Avoid double encoding to JSON in InferenceRecommenderMixin

### PR DESCRIPTION
**Pull request by SSRraymond**
*Issue #, if available:*

*Description of changes:*
Avoid double encoding to JSON

*Testing done:*
unit tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/inference_recommender/inference_recommender_mixin.py [hunk 1]
```diff
@@ -465,7 +465,7 @@ def _convert_to_endpoint_configurations_json(
             for instance_type in instance_types:
                 parameter_ranges = []
                 for name, param in parameter_range.items():
-                    as_json = param.as_json_range(name)
+                    as_json = {"Name": name, "Values": param.values}
```
Comment (jbarz1) [inline src/sagemaker/inference_recommender/inference_recommender_mixin.py:468]: add or modify existing test to ensure this doesn't get broken?

File: src/sagemaker/inference_recommender/inference_recommender_mixin.py [hunk 2]
```diff
@@ -465,7 +465,7 @@ def _convert_to_endpoint_configurations_json(
...
                     as_json["Value"] = as_json.pop("Values")
```
Comment (jbarz1) [inline src/sagemaker/inference_recommender/inference_recommender_mixin.py:469]: I think you don't need this line if you just use "Value" in line above?

Comment (SSRraymond) [reply src/sagemaker/inference_recommender/inference_recommender_mixin.py:469]: Do you mean something like:
```
as_json = {"Name": name, "Value": param.values}
```
and remove line 469?

**Review by gwang111 — APPROVED** [review]

**Review by jbarz1 — APPROVED** [review]
LGTM

**Review by trajanikant — COMMENTED** [review]
/bot run all

**Review by jinpengqi — APPROVED** [review]

File: tests/unit/sagemaker/inference_recommender/test_inference_recommender_mixin.py [hunk 3]
```diff
@@ -603,7 +650,7 @@ def test_deploy_right_size_async_override(sagemaker_session, default_right_sized
         async_inference_config=async_inference_config,
     )
 
-    assert sagemaker_session.endpoint_from_production_variants.called_with(
+    sagemaker_session.endpoint_from_production_variants.called_with(
```
Comment (jbarz1) [inline tests/unit/sagemaker/inference_recommender/test_inference_recommender_mixin.py:653]: nit: `assert_called_with` ?

**Review by navinsoni — COMMENTED** [review]
/bot run all

**Comment by SSRraymond** [comment]
Working on fix of slow tests

**Review by trajanikant — COMMENTED** [review]
/bot run all

**Review by claytonparnell — COMMENTED** [review]
/bot run slow-tests

**Comment by SSRraymond** [comment]
Regarding the slow test failure in previous build:

```
scheduled_explainability_monitor = <sagemaker.model_monitor.clarify_model_monitoring.ModelExplainabilityMonitor object at 0x7f344b435e10>
sagemaker_session = <sagemaker.session.Session object at 0x7f3450212b90>
endpoint_name = 'clarify-xgb-integ-1678224286-b86a'
ground_truth_input = 's3://sagemaker-us-west-2-142577830533/clarify-xgb-integ-1678224286-b86a/IntegTestGroundtruth'
upload_actual_data = None

    @pytest.mark.slow_test
    @pytest.mark.skipif(
        tests.integ.test_region() in tests.integ.NO_MODEL_MONITORING_REGIONS,
        reason="ModelMonitoring is not yet supported in this region.",
    )
    def test_run_explainability_monitor(
        scheduled_explainability_monitor,
        sagemaker_session,
        endpoint_name,
        ground_truth_input,
        upload_actual_data,
    ):
>       _verify_execution_status(scheduled_explainability_monitor)

tests/integ/test_clarify_model_monitor.py:410: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

monitor = <sagemaker.model_monitor.clarify_model_monitoring.ModelExplainabilityMonitor object at 0x7f344b435e10>

    def _verify_execution_status(monitor):
        _wait_for_completion(monitor)
        executions = monitor.list_executions()
        assert len(executions) > 0
        schedule_desc = monitor.describe_schedule()
        execution_summary = schedule_desc.get("LastMonitoringExecutionSummary")
        last_execution_status = execution_summary["MonitoringExecutionStatus"]
>       assert last_execution_status in ["Completed", "CompletedWithViolations"]
E       AssertionError: assert 'Failed' in ['Completed', 'CompletedWithViolations']
```

We ran the integration test locally with the following command: `tox tests/integ/test_clarify_model_monitor.py::test_run_explainability_monitor` and it is successful
Screenshot 2023-03-07 at 5 19 02 PM

**Review by trajanikant — COMMENTED** [review]
/bot run all

**Comment by SSRraymond** [comment]
The following slow tests are failing:
```
FAILED tests/integ/test_inference_pipeline.py::test_inference_pipeline_model_deploy_and_update_endpoint
FAILED tests/integ/test_clarify_model_monitor.py::test_run_explainability_monitor
FAILED tests/integ/test_clarify_model_monitor.py::test_run_bias_monitor - sto...
```

We ran the tests locally with the command:
`tox tests/integ/test_inference_pipeline.py::test_inference_pipeline_model_deploy_and_update_endpoint`
`tox tests/integ/test_clarify_model_monitor.py::test_run_bias_monitor` 
and the result is successful

Screenshot 2023-03-08 at 7 53 28 AM
Screenshot 2023-03-08 at 8 14 06 AM

`tests/integ/test_clarify_model_monitor.py::test_run_explainability_monitor` was tested in previous comment

**Review by trajanikant — COMMENTED** [review]
/bot run pr, slow-tests

**Comment by SSRraymond** [comment]
```
ERROR tests/integ/test_airflow_config.py - Exception: 'airflow import ' has r...
ERROR tests/integ/test_airflow_config.py - Exception: 'airflow import ' has r...
ERROR gw2
ERROR gw1
```
`test_airflow_config.py` tests being flaky is a known issue.

**Review by trajanikant — COMMENTED** [review]
/bot run slow-tests

**Comment by SSRraymond** [comment]
`FAILED tests/integ/test_inference_pipeline.py::test_inference_pipeline_model_deploy_and_update_endpoint`
Testes in previous comment and successful.

**Pull request merged**